### DISCUSSION TITLE
Vertically connected hulls are not considered connected for purposes of fire detection.

### DISCUSSION THREAD

**Discussion by BlackMoons** [FakeFishGames/Barotrauma#13801] [Bug reports]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [ ] My issue happened while using mods.

### What happened?

When using smoke detectors, two hulls, one over the other do not count as 'connected'

Take for example the following:

![image](https://github.com/FakeFishGames/Barotrauma/assets/15382398/e6d68220-f223-45f9-bc3a-611618468c83)

when a fire is spawned in the lower hull, you get the following:

![image](https://github.com/FakeFishGames/Barotrauma/assets/15382398/90b0c2d9-b208-403a-ac0c-0635d7e4c112)

Where the detectors in the room to the left detect the fire just fine, but the ones in the hull above the fire don't detect anything until that hull catches on fire.

### Reproduction steps

[Firebug.zip](https://github.com/FakeFishGames/Barotrauma/files/15053117/Firebug.zip)

Open submarine, use ambientlighting 0.5,0.5,0.5,0.5 so you can see, use fire command to spawn fire.

Observe that upper smoke detector lights do not turn on until the hull they are in catches fire. 

### Bug prevalence

Happens every time I play

### Single player or multiplayer?

Single player

### Version

v1.3.0.4

### Which operating system did you encounter this bug on?

Windows

**Comment by BlackMoons** [comment]
Additional testing:

![image](https://github.com/FakeFishGames/Barotrauma/assets/15382398/9d1c98e0-242c-4d91-a8d3-6dd64e749f65)

Expected side hulls to fully light up when I lit the center lower hull on fire... What actually happened was (Incorrect behavior):

![image](https://github.com/FakeFishGames/Barotrauma/assets/15382398/ee96e04f-8541-44ce-a0ed-2ae007635109)

Looks like smoke detectors don't detect fire if the hull the fire is in, is lower then the position of the smoke detector. 

When the fire spread into the upper center hull, this happened (Correct behavior):

![image](https://github.com/FakeFishGames/Barotrauma/assets/15382398/af1014d4-56a6-44f1-81c5-f3bc6ff09d0c)


And setting a fire in the right hull resulted in (Correct behavior):

![image](https://github.com/FakeFishGames/Barotrauma/assets/15382398/ba1b45f5-5c45-4d05-bdd8-03b7f7c1cd9a)

**Comment by SomeRandomNoobKekeke** [comment]
I don't get what you expect to be the correct behavior, e.g. why on last screen 3 left lights are off? why is this correct?

Like, did you mean all lights in all connected hulls should turn on instantly?
Or all lights in a square around fire?
Or you did mean they should detect smoke like in real life?
https://www.youtube.com/watch?v=ACjtIa5MU5M

**Comment by SomeRandomNoobKekeke** [comment]
> Looks like smoke detectors don't detect fire if the hull the fire is in, is lower then the position of the smoke detector.

hmm...
https://github.com/FakeFishGames/Barotrauma/blob/master/Barotrauma/BarotraumaShared/SharedSource/Items/Components/Signal/SmokeDetector.cs#L71

https://github.com/FakeFishGames/Barotrauma/blob/master/Barotrauma/BarotraumaShared/SharedSource/Map/FireSource.cs#L342

Yea, i think because of this:  `worldPosition.Y > hull.WorldRect.Y) return false;`

So it's damaging everything from top to bottom of its hull, otherwise it will damage items through walls

**Comment by BlackMoons** [comment]
Correct behavior should be that all lights within 5M of the fire, within connected hulls should be on. 

In the last pic the left most lights don't light up because they are further away then 5m, but all the lights in the square do.

Its incorrect that the 'connected hull' above the middle hull doesn't light up when the fire is in the center hull below it. Its correct when fire is to the left or right hull.
